### ISSUE TITLE
Feature request: .bind() method so that we can deconstruct instance methods while preserving internal context

### ISSUE THREAD

**Issue by samueljoli** [hapipal/schmervice#1]
It would be nice if we could deconstruct methods from a schmervice instance and be able to use them while preserving `this` context. I figured it could work much like Joi's [bind](https://github.com/hapijs/joi/blob/master/API.md#bind) method. One downside being that getter methods would not work.

```js
const serviceX = server.services();
const { transform } = serviceX.bind();

const mappedDocument = transform(payload);
```

**Comment by devinivy** [comment]
I'm interested!  The precedent set by joi is a great reference 👍 Why would getters not work?  One other thought is that it might be nice if the result of `bind()` were cached, since it would likely be in the hot path of route handlers, etc.

**Comment by samueljoli** [comment]
@devinivy Oops I misspoke about getters not working! I'll go ahead and open a pr.

**Comment by devinivy** [comment]
Closed by #2 and released in v1.2.0.  Thanks, @Samueljoli!
